builtins/match.go

```diff
@@ -34,12 +38,76 @@ func matchRecord(a, b *runtime.Record) (bool, error) {
 			}
 		} else {
 			return false, runtime.Errorf("Value mismatch: field %s in second Record not found in first", k)
-
 		}
 	}
 	return true, nil
 }
 
+func matchRecordOf(a, b *runtime.List) (bool, error) {
+	/*
+	 * Backtrack to previous * on mismatch and retry starting one
+	 * object later in the list.  Because * matches all objects
+	 * (no exception for /), it can be easily proven that there's
+	 * never a need to backtrack multiple levels.
+	 */
+	var back_pat, back_str = -1, -1
+	str := a.Elements
+	pat := b.Elements
+
+	/*
+	 * Loop over each object in pat, matching
+	 * it against the remaining unmatched tail of str.  Return false
+	 * on mismatch, or true after matching the trailing nul bytes.
+	 */
+	for i, j := 0, 0; ; {
+		var (
+			aContinues                = len(str) > i
+			bContinues                = len(pat) > j
+			c, d       runtime.Object = runtime.Undefined, runtime.Undefined
+		)
+		if aContinues {
+			c = str[i]
+		}
+		if bContinues {
+			d = pat[j]
+		}
```

Comment:
It's considered good style to have a uniform declarations. Also it would help readbility to get rid of `aContinues` and `bContinues`. How do you find something like this:
```go
str, i, back_i := a.Elements, 0, -1
pat, j, back_j := b.Elements, 0, -1
for i < len(str) && j < len(pat) {
    s := str[i]
    p := pat[j]
    switch {
    case p == runtime.Any:
        i++
        if i == len(s) {
             return false
        }
        ...